crypto/src/main/scala/coop/rchain/crypto/codec/Base16.scala

```diff
@@ -3,13 +3,13 @@ package coop.rchain.crypto.codec
 object Base16 {
   def encode(input: Array[Byte]): String = bytes2hex(input, None)
 
-  def decode(input: String): Array[Byte] =
-    if (input.length % 2 == 0) {
-      hex2bytes(input)
-    } else {
-      // TODO: Learn how to handle errors more functionally
-      throw new Error("Input must always be even in length")
-    }
+  def decode(input: String): Array[Byte] = {
+    val paddedInput = 
+      if (input.length % 2 == 0) input
+      else "0" + input
+
+    hex2bytes(paddedInput)
+  }
```

Comment:
Do we have a round trip test for encode/decode?

Reply:
now we have. See the last commit